packages/cli/src/fingerprint/fingerprintWatchCommand.js

```diff
@@ -7,30 +9,41 @@ class FingerprintWatchCommand {
     this.directory = directory;
     this.print = true;
     this.numProcessed = 0;
+    this.pidfilePath =
+      process.env.APPMAP_WRITE_PIDFILE &&
+      path.join(this.directory, 'index.pid');
   }
 
-  execute() {
-    if (verbose()) {
-      console.warn(`Watching appmaps in ${this.directory}`);
+  removePidfile() {
+    if (this.pidfilePath) {
+      console.log(`Removing ${this.pidfilePath}`);
+      fs.removeSync(this.pidfilePath);
     }
+  }
 
-    this.fpQueue = new FingerprintQueue();
-    this.fpQueue.setCounterFn(() => {
-      this.numProcessed += 1;
-    });
-    this.fpQueue.process();
-    this.watcher = chokidar.watch(`${this.directory}/**/*.appmap.json`, {
-      ignoreInitial: true,
+  execute() {
+    return new Promise((resolve) => {
+      this.fpQueue = new FingerprintQueue();
+      this.fpQueue.setCounterFn(() => {
+        this.numProcessed += 1;
+      });
+      this.fpQueue.process();
+      this.watcher = chokidar.watch(`${this.directory}/**/*.appmap.json`, {
+        ignoreInitial: true,
+      });
+      this.watcher
+        .on('add', this.added.bind(this))
+        .on('change', this.changed.bind(this))
+        .on('unlink', this.removed.bind(this))
+        .on('ready', this.ready.bind(this, resolve));
     });
-    this.watcher
-      .on('add', this.added.bind(this))
-      .on('change', this.changed.bind(this))
-      .on('unlink', this.removed.bind(this));
   }
 
-  close() {
-    this.watcher.close();
+  async close() {
+    await this.watcher.close();
+    this.removePidfile();
     this.watcher = null;
+    this.pidfilePath = null;
```

Comment:
Would you want to do this in removePidFile?

Reply:
Oops, yeah.